experiment/builder_runner.py

```diff
@@ -376,12 +378,19 @@ def build_and_run(self, generated_project: str, target_path: str,
                cwd=oss_fuzz_checkout.OSS_FUZZ_DIR)
         break
       except sp.CalledProcessError as e:
-        # Remove \n for single log entry on cloud.
+        # Replace \n for single log entry on cloud.
         stdout = e.stdout.decode('utf-8').replace('\n', '\t')
         stderr = e.stderr.decode('utf-8').replace('\n', '\t')
-        if (any(error in stdout + stderr for error in self._RETRYABLE_ERRORS)
-            and attempt_id < CLOUD_EXP_MAX_ATTEMPT):
+
+        captured_error = next(
+            (err for err in self._RETRYABLE_ERRORS if err in stdout + stderr),
+            '')
+        if captured_error and attempt_id < CLOUD_EXP_MAX_ATTEMPT:
           delay = 5 * 2**attempt_id
+          if captured_error == 'RESOURCE_EXHAUSTED':
+            # Add random jitter in case of exceeding request per minute quota.
+            delay += random.randint(1, 10)
```

Comment:
Given the quota is `per minute`, would it be better to wait at least 60 secs?

Reply:
If long waiting time is acceptable in this case. The rate is calculated in a 60 sec moving window per [Compute Engine rate quotas](https://cloud.google.com/compute/api-quota#:~:text=Google%20enforces%20rate%20quotas%20at%20a%20per%2Dminute%20(60%20seconds)%20interval%20for%20each%20group.).

Reply:
Yes, waiting a few minutes is acceptable because this error happens when we exceed the quota limit, which only happens in large-scale experiments.

Reply:
> The rate is calculated in a 60 sec moving window per [Compute Engine rate quotas](https://cloud.google.com/compute/api-quota#:~:text=Google%20enforces%20rate%20quotas%20at%20a%20per%2Dminute%20(60%20seconds)%20interval%20for%20each%20group.).

IIUC, given `attempt_id` won't be more than 5, the current delay will be at most 2.83 minutes (=5*2**5+10 seconds).
In weekly experiments with the `all` benchmark set, we have ~1500 benchmarks * 10 fuzz targets per benchmark.
More than 3 minutes is likely required for this experiment size.